v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/BigtableChangeStreamsToPubSub.java

```diff
@@ -0,0 +1,492 @@
+
+    PCollection<FailsafeElement<String, String>> nonRetryableDlqModJsonFailsafe =
+        dlqModJson.get(DeadLetterQueueManager.PERMANENT_ERRORS).setCoder(FAILSAFE_ELEMENT_CODER);
+    LOG.info(
+        "DLQ manager severe DLQ directory with date time: {}",
+        dlqManager.getSevereDlqDirectoryWithDateTime());
+    LOG.info("DLQ manager severe DLQ directory: {}", dlqManager.getSevereDlqDirectory() + "tmp/");
+    nonRetryableDlqModJsonFailsafe
+        .apply(
+            "Write Mod JSON With Non-retryable Error To DLQ",
+            MapElements.via(new StringDeadLetterQueueSanitizer()))
+        .setCoder(StringUtf8Coder.of())
+        .apply(
+            DLQWriteTransform.WriteDLQ.newBuilder()
+                .withDlqDirectory(dlqManager.getSevereDlqDirectoryWithDateTime())
+                .withTmpDirectory(dlqManager.getSevereDlqDirectory() + "tmp/")
+                .setIncludePaneInfo(true)
+                .build());
+
+    return pipeline.run();
+  }
+
+  private static Instant toInstant(Timestamp timestamp) {
+    if (timestamp == null) {
+      return null;
+    } else {
+      return Instant.ofEpochMilli(timestamp.getSeconds() * 1000 + timestamp.getNanos() / 1000000);
+    }
+  }
+
+  private static DeadLetterQueueManager buildDlqManager(
+      BigtableChangeStreamsToPubSubOptions options) {
+    String tempLocation =
+        options.as(DataflowPipelineOptions.class).getTempLocation().endsWith("/")
+            ? options.as(DataflowPipelineOptions.class).getTempLocation()
+            : options.as(DataflowPipelineOptions.class).getTempLocation() + "/";
+    String dlqDirectory =
+        options.getDlqDirectory().isEmpty() ? tempLocation + "dlq/" : options.getDlqDirectory();
+
+    LOG.info("DLQ directory: {}", dlqDirectory);
+    return DeadLetterQueueManager.create(dlqDirectory, options.getDlqMaxRetries());
+  }
+
+  private static String getBigtableCharset(BigtableChangeStreamsToPubSubOptions options) {
+    return StringUtils.isEmpty(options.getBigtableChangeStreamCharset())
+        ? "UTF-8"
+        : options.getBigtableChangeStreamCharset();
+  }
+
+  private static String getBigtableProjectId(BigtableChangeStreamsToPubSubOptions options) {
+    return StringUtils.isEmpty(options.getBigtableReadProjectId())
+        ? options.getProject()
+        : options.getBigtableReadProjectId();
+  }
+
+  private static String getPubSubProjectId(BigtableChangeStreamsToPubSubOptions options) {
+    return StringUtils.isEmpty(options.getPubSubProjectId())
+        ? options.getProject()
+        : options.getPubSubProjectId();
+  }
+
+  private static Boolean validateSchema(
+      BigtableChangeStreamsToPubSubOptions options, PubSubUtils pubSub) throws Exception {
+    Topic topic = pubSub.getDestination().getPubSubTopic();
+    if (options.getMessageEncoding() != topic.getSchemaSettings().getEncoding().toString()) {
+      return false;
+    }
+    String messageFormatPath = topic.getSchemaSettings().getSchema();
```

Comment:
It appears as getSchemaSettings() can return null:  https://screenshot.googleplex.com/BpgpWgZP95JHPYD 

If so, we should return true